product_last_purchase_date/models/product_product.py

```diff
@@ -38,3 +38,10 @@ def _compute_last_purchase_date(self):
                 last_purchase_date = man_last_purchase_date
             product.last_purchase_date = last_purchase_date
             product.product_tmpl_id._compute_last_purchase_date()
+
+    def _get_domain(self, product):
```

Comment:
```suggestion
    def _get_move_domain(self, product):
```
Just to reduce the chances of the method name clash...

We usually have place the method above the caller, for better readability.